packages/@lwc/synthetic-shadow/src/faux-shadow/focus.ts

```diff
@@ -102,11 +114,35 @@ interface QuerySegments {
     next: HTMLElement[];
 }
 
+export function hostElementFocus(this: HTMLElement) {
+    // Open an issue against TypeScript and add an LWC issue here
+    const rootNode = (this.getRootNode() as unknown) as DocumentOrShadowRoot;
```

Comment:
this should be fixed right? it might not be a shadowRoot, or are you assuming that the `this` node is connected?

Reply:
this is called by the focus routine, which might be fine... still I think we should at least add an assertion/error or whatever there in dev-mode if the thing is not connected, just to avoid issues.

Reply:
@caridy It feels like an assertion would be overkill since invoking the focus() method on a disconnected element isn't necessarily an issue, but we can add it there if you think it might be valuable. What do you think?